### PR TITLE
LS-191 Upgrade to Recent Felix

### PR DESCRIPTION
**Pull request by krinnewitz**
This updates most of lecturesight's dependencies to their latest versions. Updating javacl was not necessary. Since javacl is more or  less abandoned, it may make sense to migrate to jocl. However, that could be quite some work.

Unfortunately, remote shell access via telnet or ssh is not supported with the modern jline shell. One workaround is to downgrade the felix shell by removing org.apache.felix.gogo.jline-1.1.0.jar and re-adding org.apache.felix.gogo.shell-1.1.2.jar as well as org.apache.felix.shell.remote-1.2.0.jar.

Since felix SCR annotations are now depricated, we now have to define xml configs instead.

### CODE REVIEW

File: src/modules/lecturesight-framesource-impl/src/main/resources/OSGI-INF/cv.lecturesight.framesource.impl.FrameSourceManagerImpl.xml [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+  <service>
+    <provide interface="cv.lecturesight.framesource.FrameSourceManager"/>
+    <provide interface="org.osgi.service.event.EventHandler"/>
+  </service>
+
+  <property name="event.topics">
+      org/osgi/framework/ServiceEvent/REGISTERED
+      org/osgi/framework/ServiceEvent/UNREGISTERING
+  </property>
+  <reference name="config"
```
Comment (smarquard) [inline src/modules/lecturesight-framesource-impl/src/main/resources/OSGI-INF/cv.lecturesight.framesource.impl.FrameSourceManagerImpl.xml]: This file could use the more compact format used elsewhere for consistency.

File: src/modules/lecturesight-ptzcontrol-onvif/pom.xml [hunk 2]
```diff
@@ -59,34 +55,10 @@
...
-					</dependency>
-				</dependencies>
-			</plugin>
-
 			<plugin>
 				<groupId>org.apache.felix</groupId>
 				<artifactId>maven-bundle-plugin</artifactId>
-				<version>2.0.1</version>
-				<extensions>true</extensions>
+        <extensions>true</extensions>
```
Comment (smarquard) [inline src/modules/lecturesight-ptzcontrol-onvif/pom.xml]: Indentation has gone a bit wonky in this file.

File: src/modules/lecturesight-ptzcontrol-vapix/pom.xml [hunk 3]
```diff
@@ -122,35 +118,10 @@
...
-				</dependencies>
-
-			</plugin>
-
 			<plugin>
 				<groupId>org.apache.felix</groupId>
 				<artifactId>maven-bundle-plugin</artifactId>
-				<version>2.0.1</version>
-				<extensions>true</extensions>
+        <extensions>true</extensions>
```
Comment (smarquard) [inline src/modules/lecturesight-ptzcontrol-vapix/pom.xml]: Indentation wonky. Maybe there's a mix of spaces and tabs here?

File: src/modules/lecturesight-framesource-impl/src/main/resources/OSGI-INF/cv.lecturesight.framesource.impl.FrameSourceManagerImpl.xml (hunk 1 above)
Comment (krinnewitz) [reply src/modules/lecturesight-framesource-impl/src/main/resources/OSGI-INF/cv.lecturesight.framesource.impl.FrameSourceManagerImpl.xml]: Thanks, fixed.

File: src/modules/lecturesight-ptzcontrol-onvif/pom.xml (hunk 2 above)
Comment (krinnewitz) [reply src/modules/lecturesight-ptzcontrol-onvif/pom.xml]: Thanks, fixed.

File: src/modules/lecturesight-ptzcontrol-vapix/pom.xml (hunk 3 above)
Comment (krinnewitz) [reply src/modules/lecturesight-ptzcontrol-vapix/pom.xml]: Thanks, fixed.

**Comment by smarquard** [comment]
This looks good to me - builds, starts up and runs correctly (I tested using gogo.shell rather than gogo.jline).

**Comment by smarquard** [comment]
Thanks @krinnewitz!

**Comment by smarquard** [comment]
Thanks @JamesUoM. Plapadoo has wrapped up work on this, so I'll fix the scripted camera operator build.

**Comment by smarquard** [comment]
Fixed with https://github.com/LectureSight/lecturesight/commit/2cbbd178336d493c23cb7a91bd442efcc4cefaf2

**Pull request merged**